Mathlib/Data/Multiset/DershowitzManna.lean

```diff
@@ -0,0 +1,411 @@
+/-
+Copyright (c) 2024 Haitian Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Haitian Wang, Malvin Gattinger
+-/
+import Mathlib.Tactic.Linarith
+import Mathlib.Data.Multiset.Basic
+import Mathlib.Logic.Relation
+
+/-!
+# Dershowitz-Manna ordering
+
+In this file we define the _Dershowitz-Manna ordering_ on multisets. Specifically, for two multisets
+`M` and `N` in a partial order `(S, <)`, `M` is smaller than `N` in the Dershowitz-Manna ordering if
+`M` can be obtained from `N` by replacing one or more elements in `N` by some finite number of
+elements from `S`, each of which is smaller (in the underling ordering over `S`) than one of the
+replaced elements from `N`. We prove that, given a well-founded partial order on the underlying set,
+the Dershowitz-Manna ordering defined over multisets is also well-founded.
+
+## Main results
+
+- `Multiset.IsDershowitzMannaLT` : the standard definition fo the `Dershowitz-Manna ordering`.
+- `Multiset.IsDershowitzMannaLT.wf` : the main theorem about the `Dershowitz-Manna ordering`
+being well-founded.
+- `Multiset.TransLT_eq_DMLT` : two definitions of the `Dershowitz-Manna ordering` are equivalent.
+
+## References
+
+* [Wikipedia, Dershowitz–Manna ordering*]
+(https://en.wikipedia.org/wiki/Dershowitz%E2%80%93Manna_ordering)
+
+* [CoLoR](https://github.com/fblanqui/color), a Coq library on rewriting theory and termination.
+  Our code here is inspired by their formalization and the theorem is called `mOrd_wf` in the file
+  [MultisetList.v](https://github.com/fblanqui/color/blob/1.8.5/Util/Multiset/MultisetOrder.v).
+
+-/
+
+namespace Multiset
+
+variable {α : Type*}
```

Comment:
```suggestion
variable {α : Type*} [Preorder α]
```
so that you don't have to add it to every single statement

Reply:
Aha, that is helpful!